src/webgl/p5.RendererGL.js

```diff
@@ -874,12 +879,43 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
     this.curStrokeJoin = join;
   }
 
-  filter(filterType) {
-    // filter can be achieved using custom shaders.
-    // https://github.com/aferriss/p5jsShaderExamples
-    // https://itp-xstory.github.io/p5js-shaders/#/
-    p5._friendlyError('filter() does not work in WEBGL mode');
+  filter(args) {
+    // Couldn't create graphics in RendererGL constructor
+    // (led to infinite loop)
+    // so it's just created here once on the initial filter call.
+    if (!this.filterGraphicsLayer) {
+      this.filterGraphicsLayer =
+        new p5.Graphics(this.width, this.height, constants.WEBGL, this._pInst);
+    }
+
+    // TODO, handle different signatures:
+    // if args[0] is a p5.Shader,
+    //   this.filterShader = args[0]
+    // else
+    //   this.filterShader = map(args[0], {GRAYSCALE: grayscaleShader, ...})
+    //   filterOperationParameter = undefined or args[1]
+
+    let userShader = args[0];
+    // Bind the shader to the pg renderer,
+    // since when the user makes it with createShader() it binds to the main
+    userShader =
+      new p5.Shader(pg._renderer, userShader._vertSrc, userShader._fragSrc);
```

Comment:
Should we provide users with a default vertex shader?

Reply:
I think so. Here are some related thoughts: #6144 
I'm leaning towards the simplest option, just exposing a default vertex shader as a string constant.

Reply:
Another thought, we could make a function to make a shader that only takes in a fragment shader and automatically uses that default vertex shader? Like createFilterShader()?

Reply:
A default vertex shader as a string constant honestly makes sense to me, although I'd like to hear other thoughts. Would the idea be that they would use createShader() like normal, but pass in the new constant?